sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/ComplexTypes.scala

```diff
@@ -25,8 +25,8 @@ import org.apache.spark.sql.catalyst.rules.Rule
 * push down operations into [[CreateNamedStructLike]].
 */
 object SimplifyCreateStructOps extends Rule[LogicalPlan] {

```

Comment:
nit: can we merge these 3 rules? then we only need to transform the plan once.

Reply:
+1 for @cloud-fan 's advice.